File: models/10.1161:01.CIR.96.1.44/example_b.py
"""
example_b.py
by Ted Morin

contains example code for 1-year Hypertension risks using Weibull betas from
10.7326/0003-4819-148-2-200801150-00005
2008 A Risk Score for Predicting Near-Term Incidence of Hypertension
Framingham Heart Study
"""

from model_b import model

'Sex' 'Age' 'Systolic BP' 'Diastolic BP' 'Cigarettes per day' 'Total Cholesterol' 'Diabetes' 'Previous CHD'

toscore = []     
toscore.append( [1,55,146,88,6,250,0,0] ) 
toscore.append( [1,70,146,89,30,240,1,0] ) 

scores = []
print "Risks"
for i in range(len(toscore)):
    scores.append( model(toscore[i][0],toscore[i][1],toscore[i][2],toscore[i][3],toscore[i][4],toscore[i][5],toscore[i][6],toscore[i][7]) )
    print "%.3f" % (float(scores[i])*100.)

File: models/10.1001:jama.290.8.1093/example_b.R
# example_b.R
# by Ted Morin
# 
# contains example code for predicting 5-year Stroke or Death Risk for patients with new-onset AF using beta coefficients from
# 10.1001/jama.290.8.1093
# 2003 A Risk Score for Predicting Stroke or Death in Individuals With New-Onset Atrial Fibrillation in the Community
#
# Warfarin censored (Table 2, far-right)
#
# function expects parameters:
#  "age"   "Systolic BP" "Diabetes Mellitus" "History of CHF or MI" "Cigarette Smoking" "Significant Murmur" "Left Ventricular Hypertrophy"
#   years        mg/dL        
#  int/float    int/float         bool                bool                 bool                bool                        bool


# load the model
load('model_b.Rdata')


# testpatient1
testpat = c(70, 120, 0, 0, 0, 0, 0)
model(testpat)
# testpatient2
testpat = c(65, 130, 0, 0, 0, 0, 0)
model(testpat)
# testpatient3
testpat = c(65, 130, 0, 1, 0, 0, 0)
model(testpat)
# testpatient4
testpat = c(65, 130, 1, 1, 0, 0, 0)
model(testpat)
# testpatient5
testpat = c(65, 130, 0, 0, 0, 1, 1)
model(testpat)